include/theme.h

```diff
@@ -30,6 +31,14 @@ struct theme_snapping_overlay {
 	float border_color[3][4];
 };
 
+enum lab_button_state {
```

Comment:
Then maybe using bitmap as the array index is not an extensible option since `pressed` is exclusive with `hover` and `disabled` is exclusive with all other states, whereas `hover`/`toggled`/`rounded` states are all in orthogonal relationship...

I still insist on using bitmap as index in this PR as it's the cleanest option at the moment, but when we implement those states in the future, maybe we need to store buffers/nodes in `wl_list` rather than an array.